hw-event-proxy/cmd/main.go

```diff
@@ -0,0 +1,224 @@
+	wg.Add(1)
+	startWebhook(&wg)
+	log.Info("waiting for events")
+	wg.Wait()
+}
+
+func createPublisher() (pub pubsub.PubSub, err error) {
+	baseURL := types.ParseURI(fmt.Sprintf("http://localhost:%d%s", apiPort, apiPath))
+	publisherURL := types.ParseURI(fmt.Sprintf("%s%s", baseURL, "publishers"))
+	returnURL := types.ParseURI(fmt.Sprintf("%s%s", baseURL, "dummy"))
+	publisher := v1pubsub.NewPubSub(returnURL, resourceAddress)
+
+	var pubB []byte
+	var status int
+	if pubB, err = json.Marshal(&publisher); err == nil {
+		rc := restclient.New()
+		if status, pubB = rc.PostWithReturn(publisherURL, pubB); status != http.StatusCreated {
+			err = fmt.Errorf("failed to create publisher creation api at %s, returned status %d", publisherURL, status)
+			return pub, err
+		}
+	} else {
+		log.Errorf("failed to marshal publisher: %v", err)
+		return pub, err
+	}
+	if err = json.Unmarshal(pubB, &pub); err != nil {
+		log.Errorf("failed to unmarshal publisher: %v", err)
+		return pub, err
+	}
+	return pub, nil
+}
+
+func startWebhook(wg *sync.WaitGroup) {
+	http.HandleFunc("/ack/event", ackEvent)
+	http.HandleFunc("/webhook", handleHwEvent)
+	// TODO: retry on fail
+	go func() {
+		defer wg.Done()
+		err := http.ListenAndServe(fmt.Sprintf(":%d", util.GetIntEnv("HW_EVENT_PORT")), nil)
+		if err != nil {
+			log.Errorf("error with webhook server %s\n", err.Error())
+		}
+	}()
+}
+
+func ackEvent(w http.ResponseWriter, req *http.Request) {
+	defer req.Body.Close()
+	bodyBytes, err := ioutil.ReadAll(req.Body)
+	if err != nil {
+		log.Errorf("error reading acknowledgment %v", err)
+	}
+	e := string(bodyBytes)
+	if e != "" {
+		log.Debugf("received ack %s", string(bodyBytes))
+	} else {
+		w.WriteHeader(http.StatusNoContent)
+	}
+}
+
+// handleHwEvent gets redfish HW events and converts it to cloud native event
+// and publishes to the event framework publisher
+func handleHwEvent(w http.ResponseWriter, r *http.Request) {
+	defer r.Body.Close()
+
+	bodyBytes, err := ioutil.ReadAll(r.Body)
+	if err != nil {
+		log.Errorf("error reading hw event: %v", err)
+		return
+	}
+	event := createHwEvent()
+	redfishEvent := hwevent.RedfishEvent{}
+	err = json.Unmarshal(bodyBytes, &redfishEvent)
+	if err != nil {
+		log.Errorf("failed to unmarshal hw event: %v", err)
+		return
+	}
+	for i, e := range redfishEvent.Events {
+		if e.Message == "" {
+			parsed, err := parseMessage(e)
+			if err == nil {
+				redfishEvent.Events[i] = parsed
+			} else {
+				// ignore error
+				log.Debugf("error parsing message: %v", err)
+			}
+		}
+	}
+
+	data := v1hwevent.CloudNativeData()
+	data.SetVersion(hwEventVersion) //nolint:errcheck
+	data.SetData(&redfishEvent)
+	event.SetData(data)
+	err = publishHwEvent(event)
+	if err != nil {
+		log.Errorf("failed to publish hw event: %v", err)
+	}
+}
+
+// TODO: add timeout or check server ready
+func parseMessage(m hwevent.EventRecord) (hwevent.EventRecord, error) {
+	addr := fmt.Sprintf("localhost:%d", util.GetIntEnv("MSG_PARSER_PORT"))
```

Comment:
MSG_PARSER_PORT, make it a constant

Reply:
This is passed from deployment yaml file so that message-parser and hw-event-proxy containers can use the same value.